ui/src/ducks/app/monitoring.js

```diff
@@ -333,7 +333,7 @@ function getClusterQueryStatus(result) {
 }
 
 export function* handlePrometheusError(clusterHealth, result) {
-  if (result.error.response) {
+  if (result?.error?.response?.status < 500) {
```

Comment:
I think you can now use `PrometheusQueryResult`(https://github.com/scality/metalk8s/blob/5d779b89c90419fa8de5508b255467673a945738/ui/src/services/prometheus/api.js#L33) to type `result` argument and that may help you to simplify optional chaining use here.

Reply:
Hi @JBWatenbergScality, agree! we should use typing to simplify the optional chaining here. 
However, enable flow typing in this monitoring reducer would introduce a lot of typing errors.
So I added TODO commit and will address later on this issue: https://github.com/scality/metalk8s/issues/3001
WDYT?

Reply:
ok